azurerm/internal/services/eventgrid/eventgrid_system_topic_resource.go

```diff
@@ -0,0 +1,217 @@
+package eventgrid
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/eventgrid/mgmt/2020-04-01-preview/eventgrid"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/eventgrid/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmEventGridSystemTopic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmEventGridSystemTopicCreateUpdate,
+		Read:   resourceArmEventGridSystemTopicRead,
+		Update: resourceArmEventGridSystemTopicCreateUpdate,
+		Delete: resourceArmEventGridSystemTopicDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.EventGridSystemTopicID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"source_resource_id": {
```

Comment:
Before it was just `source`, but was requested by @katbyte to be changed to `source_resource_id`. It's the Azure Resource ID of the event emitting resource.

Reply:
Shall we call it `source_id`?

Reply:
@jrauschenbusch apologies, busy week - after chatting with @katbyte about this could we make this `source_arm_resource_id`, which should solve this & avoid that confusion?

Reply:
Alright. Maybe this naming pattern should be applied all the time in the future for the terraform azurerm resources, when they have to reference IDs of ARM resources.

Reply:
Done. Changed to `source_arm_resource_id`.